packages/editor/src/components/combo-box/combo-box.ts

```diff
@@ -0,0 +1,173 @@
+import type { ComboBoxOption } from './types.ts';
+
+@customElement('clippy-combo-box')
+export class ComboBox extends LitElement {
+  @property({ type: Boolean }) disabled = false;
+  @property({ type: Boolean }) readOnly = false;
+  @property({ type: Number }) private activeIndex = -1;
+  @property({ type: Boolean }) private isOpen = false;
+  @property({ type: Array }) private options: ComboBoxOption[] = [];
+  @property({ type: String }) value = '';
+  @property({ type: Function }) onSelect = (value: string) => value;
+
+  override willUpdate(changedProperties: PropertyValues) {
+    if (changedProperties.has('options')) {
+      const activeOption = this.options.find((option) => option.active);
+      if (activeOption) {
+        this.value = activeOption.label;
+      }
+    }
+  }
+
+  static override readonly styles = [
+    css`
+      :host {
+        position: relative;
+        display: inline-block;
+      }
+      .clippy-combo-box {
+        inline-size: 100%;
+        max-inline-size: 500px;
+        position: relative;
+      }
+      .clippy-combobox__input {
+        line-height: 1.8em;
+        background-color: #fff;
+        border: 1px solid var(--basis-color-action-1-color-active);
+        border-radius: 3px;
+        color: var(--basis-color-action-1-color-active);
+        padding-block: var(--basis-space-inline-sm);
+        padding-inline: var(--basis-space-inline-md);
+      }
+    `,
+  ];
+
+  private handleInput(e: Event) {
+    const input = e.target as HTMLInputElement;
+    this.value = input.value.toLowerCase();
+    this.isOpen = true;
+    this.activeIndex = -1;
+  }
+
+  private handleKeyDown(e: KeyboardEvent) {
+    if (this.disabled || this.readOnly) return;
+
+    const filteredCount = this.filteredOptions.length;
+
+    switch (e.key) {
+      case 'ArrowDown':
+        e.preventDefault();
+        this.isOpen = true;
+        this.activeIndex = this.activeIndex < filteredCount - 1 ? this.activeIndex + 1 : 0;
+        this.updateActiveDescendant(); // TODO: needed?
+        break;
+
+      case 'ArrowUp':
+        e.preventDefault();
+        this.isOpen = true;
+        this.activeIndex = this.activeIndex > 0 ? this.activeIndex - 1 : filteredCount - 1;
+        this.updateActiveDescendant();
+        break;
+
+      case 'Enter':
+        e.preventDefault();
+        if (this.activeIndex >= 0 && this.activeIndex < filteredCount) {
+          this.selectOption(this.filteredOptions[this.activeIndex]);
+        }
+        break;
+
+      case 'Escape':
+        e.preventDefault();
+        this.isOpen = false;
+        this.activeIndex = -1;
+        break;
+
+      case 'Home':
+        e.preventDefault();
+        this.activeIndex = 0;
+        this.updateActiveDescendant();
+        break;
+
+      case 'End':
+        e.preventDefault();
+        this.activeIndex = filteredCount - 1;
+        this.updateActiveDescendant();
+        break;
+    }
```

Comment:
Ik zou kijken of je deze in losse functies kan vatten zodat je meer zoiets krijgt:

```ts
const setSelection = (index, isOpen) => { /*...*/ }

switch(e.key) {
  case 'ArrowDown': return setSelection(index + 1, true)
  case 'ArrowUp': return setSelection(index - 1, true);
  case 'Enter': return commitSelection(index);
  case 'Escape' : return setSelection(-1);
  case 'Home': return setSelection(0);
  case 'End': return setSelection(count - 1);
}